src/cli/self_update/unix.rs

```diff
@@ -164,3 +164,17 @@ fn remove_legacy_paths() -> Result<()> {
     }
     Ok(())
 }
+
+fn remove_legacy_paths() -> Result<()> {
+    // Before the work to support more kinds of shells, we always inserted this
+    // which is now considered legacy
```

Comment:
I personally prefer more precise comments, because coming from outside some months later I won't know what "before the work to support" is or when it was done. If possible I would write "Before/Until version XYZ ...".

Reply:
I agree, mentioned 1.23.0